bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/AbstractConfiguration.java

```diff
@@ -204,7 +204,8 @@ public String getMetadataServiceUri() throws ConfigurationException {
             String ledgerManagerType;
             Class<? extends LedgerManagerFactory> factoryClass = getLedgerManagerFactoryClass();
             if (factoryClass == null) {
-                ledgerManagerType = HierarchicalLedgerManagerFactory.NAME;
+                // set the ledger manager type to "null", so the driver implementation knows that the type is not set.
+                ledgerManagerType = "null";
```

Comment:
Wouldn't it more user friendly something like 'auto' instead of 'null'.
Like zk+auto://.....
Instead of
zk+null://

Or maybe I did not understand correctly

Reply:
"zk+null" comes from bc + testing and used internally when `metadataServiceUri` is not provided. by default user should "zk://..." for the purpose of "auto". I don't want to introduce a concept called "auto", which is confusing.

Reply:
@eolivelli if you feel strong about "zk+null" for the internal usage,  I can change in a subsequent PR. merging this PR now to unblock subsequent changes.

Reply:
@sijie  totally makes sense.